dwl.c

```diff
@@ -1676,23 +1673,17 @@ rendermon(struct wl_listener *listener, void *data)
 
 	/* Render if no XDG clients have an outstanding resize and are visible on
 	 * this monitor. */
-	/* Checking m->un_map for every client is not optimal but works */
-	wl_list_for_each(c, &clients, link) {
-		if ((c->resize && m->un_map) || (c->type == XDGShell
-				&& (c->surface.xdg->pending.geometry.width !=
-				c->surface.xdg->current.geometry.width
-				|| c->surface.xdg->pending.geometry.height !=
-				c->surface.xdg->current.geometry.height))) {
-			/* Lie */
-			wlr_surface_send_frame_done(client_surface(c), &now);
+	wl_list_for_each(c, &clients, link)
+		if (!c->isfloating && c->resize) {
 			skip = 1;
+			/* Lie */
+			wlr_scene_output_send_frame_done(m->scene_output, &now);
 		}
-	}
-	if (!skip && !wlr_scene_output_commit(m->scene_output))
+
+	if (skip || !wlr_scene_output_commit(m->scene_output))
```

Comment:
I see no reason to change this.

Reply:
Yes because there actually isn't. I removed it and will later squash it